examples/window/camera.rs

```diff
@@ -0,0 +1,45 @@
+use dotrix::{
+    ecs::{ Const, Mut },
+    input::{ Button, State as InputState },
+    math::{ Vec2 },
+    services::{ Camera, Frame, Input, Window },
+    systems::camera_control,
+};
+
+pub struct CameraMemory {
+    cursor_position: Vec2,
+}
+
+impl CameraMemory {
+    pub fn new() -> Self {
+        Self {
+            cursor_position: Vec2::new(0.0, 0.0)
+        }
+    }
+}
+
+/// This func updates camera based on controls
+pub fn camera_update(
+    camera: Mut<Camera>,
+    frame: Const<Frame>,
+    input: Const<Input>,
+    mut memory: Mut<CameraMemory>,
+    mut window: Mut<Window>,
+) {
+
+    if let Some(state) = input.button_state(Button::MouseRight) {
+        match state {
+            InputState::Activated => {
+                memory.cursor_position = *input.mouse_position().unwrap(); // TODO: why option?
```

Comment:
Why we returning `input.mouse_position` as `Option`? Why not simply `Vec3`?

Reply:
Because on start of the app we had no information about mouse position and considered  (0, 0), which gives unwanted delta when mouse enters the window.

Reply:
I changed this behaviour. Delta is now **not** calculated from the last mouse position. We now use winit MouseMotion event, so mouse_position have no effect on delta.
We don't have information about mouse position when the cursor leaves the window, but the option doesn't solves that problem